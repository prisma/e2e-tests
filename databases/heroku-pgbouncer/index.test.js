const {
  clientWithoutQueryStringParamCall,
  clientWithQueryStringParamCall,
  client,
  clientWithQueryStringParam,
} = require('.')

describe('should test Prisma client and PgBouncer', () => {
  afterAll(async () => {
    await client.$disconnect()
    await clientWithQueryStringParam.$disconnect()
    return
  })

  it.skip('should fail with default heroku pgbouncer without the pgbouncer query string param', async () => {
    try {
      await clientWithoutQueryStringParamCall()

      /*
       * Query engine instance names prepared statements serially s0, s1 and so on. Without the `pgbouncer=true` flag,
       * prepared statements are not cleaned up in PgBouncer. By doing disconnect/reconnect, we get a
       * new instance of query engine that starts again at s0. And we expect the next client call to throw
       * "prepared statement s0 already exists"
       */
      await client.$disconnect()
      await client.$connect()

      await clientWithoutQueryStringParamCall()
      expect(1).toEqual(0) // The code should never reach here
    } catch (e) {
      expect(e).toMatchSnapshot()
    }
  })

  it('should work with default heroku pgbouncer with the pgbouncer query string param', async () => {
    const data = await clientWithQueryStringParamCall()
    expect(data).toMatchSnapshot()

    const data1 = await clientWithQueryStringParamCall()
    expect(data1).toMatchSnapshot()
  })
})
