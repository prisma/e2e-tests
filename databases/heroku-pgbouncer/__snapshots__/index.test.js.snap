// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test Prisma client and PgBouncer should fail with default heroku pgbouncer without the pgbouncer query string param 1`] = `
[Error: 
Invalid \`prisma.user.findMany()\` invocation:


  Error occurred during query execution:
ConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(Error { kind: Db, cause: Some(DbError { severity: "ERROR", parsed_severity: Some(Error), code: SqlState("42P05"), message: "prepared statement \\"s0\\" already exists", detail: None, hint: None, position: None, where_: None, schema: None, table: None, column: None, datatype: None, constraint: None, file: Some("prepare.c"), line: Some(466), routine: Some("StorePreparedStatement") }) }) })]
`;

exports[`should test Prisma client and PgBouncer should work with default heroku pgbouncer with the pgbouncer query string param 1`] = `[]`;

exports[`should test Prisma client and PgBouncer should work with default heroku pgbouncer with the pgbouncer query string param 2`] = `[]`;
